pkg/controller/jaeger/jaeger_controller.go

```diff
@@ -97,26 +99,35 @@ func (r *ReconcileJaeger) Reconcile(request reconcile.Request) (reconcile.Result
 
 	// wait for all the dependencies to succeed
 	if err := r.handleDependencies(str); err != nil {
+		log.WithFields(log.Fields{
+			"namespace": instance.Namespace,
+			"instance":  instance.Name,
+		}).WithError(err).Error("failed to handle the dependencies")
 		return reconcile.Result{}, err
 	}
 
-	created, err := r.handleCreate(str)
+	applied, err := r.apply(*instance, str)
 	if err != nil {
-		log.WithField("instance", instance).WithError(err).Error("failed to create")
+		log.WithFields(log.Fields{
+			"namespace": instance.Namespace,
+			"instance":  instance.Name,
+		}).WithError(err).Error("failed to apply the changes")
 		return reconcile.Result{}, err
 	}
 
-	if created {
-		log.WithField("name", instance.Name).Info("Configured Jaeger instance")
-	}
-
-	if err := r.handleUpdate(str); err != nil {
-		return reconcile.Result{}, err
+	if applied {
+		log.WithFields(log.Fields{
+			"namespace": instance.Namespace,
+			"instance":  instance.Name,
+		}).Info("Configured Jaeger instance")
 	}
 
 	// we store back the changed CR, so that what is stored reflects what is being used
 	if err := r.client.Update(context.Background(), instance); err != nil {
-		log.WithError(err).Error("failed to update")
+		log.WithFields(log.Fields{
+			"namespace": instance.Namespace,
+			"instance":  instance.Name,
+		}).WithError(err).Error("failed to update")
```

Comment:
Does this message need to be more specific  - i.e. failed to store updated CR ? - as otherwise implies failed to update the jaeger instance (i.e. deployment)